pylearn2/datasets/dataset.py

```diff
@@ -73,6 +77,79 @@ def iterator(self, mode=None, batch_size=None, num_batches=None,
         # can be handled here.
         raise NotImplementedError()
 
+    def _init_iterator(self, mode=None, batch_size=None, num_batches=None,
+                       rng=None, data_specs=None):
+        """
+        Return an iterator for this dataset with the specified
+        behaviour. Unspecified values are filled-in by the default.
+
+        Parameters
+        ----------
+        mode : str or object, optional
+            One of 'sequential', 'random_slice', or 'random_uniform',
+            *or* a class that instantiates an iterator that returns
+            slices or index sequences on every call to next().
+        batch_size : int, optional
+            The size of an individual batch. Optional if `mode` is
+            'sequential' and `num_batches` is specified (batch size
+            will be calculated based on full dataset size).
+        num_batches : int, optional
+            The total number of batches. Unnecessary if `mode` is
+            'sequential' and `batch_size` is specified (number of
+            batches will be calculated based on full dataset size).
+        rng : int, object or array_like, optional
+            Either an instance of `numpy.random.RandomState` (or
+            something with a compatible interface), or a seed value
+            to be passed to the constructor to create a `RandomState`.
+            See the docstring for `numpy.random.RandomState` for
+            details on the accepted seed formats. If unspecified,
+            defaults to using the dataset's own internal random
+            number generator, which persists across iterations
+            through the dataset and may potentially be shared by
+            multiple iterator objects simultaneously (see "Notes"
+            below).
+        data_specs : tuple
+            A `(space, source)` tuple that specifies the target space and
+            source that the iterator should provide. See :ref:`data_specs` i
+            for a full description.
+
+        Returns
+        -------
+        iter_obj : object
+            An iterator object implementing the standard Python
+            iterator protocol (i.e. it has an `__iter__` method that
+            return the object itself, and a `next()` method that
+            returns results until it raises `StopIteration`).
+        """
+        # TODO How much of this is possible to put in the Dataset.iterator or
+        # in FiniteDatasetIterator?
```

Comment:
I am using it in HDF5.py too, so unless it is wrong to initialize HDF5.iterator in the same way I guess it is better to have it in Dataset rather than having duplicated code.